src/main/java/org/typesense/api/ApiCall.java

```diff
@@ -61,7 +61,7 @@ boolean isDueForHealthCheck(Node node) {
     // Loops in a round-robin fashion to check for a healthy node and returns it
     Node getNode() {
         if (configuration.nearestNode != null) {
-            if (isDueForHealthCheck((configuration.nearestNode)) || configuration.nearestNode.isHealthy) {
+            if (configuration.nearestNode.isHealthy || isDueForHealthCheck((configuration.nearestNode)) ) {
```

Comment:
The reason for having `isDueForHealthCheck((configuration.nearestNode))` first is because we want to add quarantined nodes back into the pool, without which we will only be checking the `nearestNode` always.

Reply:
Should we then also change the behavior here?
https://github.com/typesense/typesense-java/blob/27bacf7937a085ede7306bd57a70d705d6c00c32/src/main/java/org/typesense/api/ApiCall.java#L76

Reply:
Actually, I think your change is fine. Either approach is okay, because isHealthy is set during an actual network call. In the case that node is healthy, we really don't have to check for healthiness.

One problem, though with the configuration.nodes loop I think is that it will always pick the first node in the list.